app/src/main/java/com/example/androidtrainingproject/products/ProductDetailsScreen.kt

```diff
@@ -66,46 +83,125 @@ fun ProductDetails() {
         Column(
             Modifier
                 .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .paint(
+                    painter = painterResource(id = R.drawable.app_background),
+                    contentScale = ContentScale.FillBounds
+                )
                 .padding(top = paddingValues.calculateTopPadding())
-                .padding(10.dp)
+                .padding(dimensionResource(id = R.dimen.padding_m))
         ) {
-            Text(
-                modifier = Modifier.fillMaxWidth(),
-                text = "Item",
-                textAlign = TextAlign.Center
-            )
-
             productDetailsViewModel.productData?.let { product ->
-                AsyncImage(
-                    model = product.image,
-                    modifier = Modifier
-                        .fillMaxWidth()
-                        .padding(10.dp)
-                        .clip(RoundedCornerShape(10.dp)),
-                    contentScale = ContentScale.Fit,
-                    alignment = Alignment.Center,
-                    contentDescription = "Item Image"
+                Box {
+                    AsyncImage(
+                        model = product.image,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .clip(RoundedCornerShape(dimensionResource(id = R.dimen.padding_sm_plus))),
+                        contentScale = ContentScale.Fit,
+                        alignment = Alignment.Center,
+                        contentDescription = stringResource(id = R.string.product_image_description)
+                    )
+                    Row(
```

Comment:
This can be exported as a reusable composable next to this file in a `layouts` package. No reason to clutter the screen composable with big chunks of custom UI except if they're not simple `Text`s or the like.

Reply:
Doesn't make sense to me to export just the row (the "In Stock" label) and leave the image outside since they are very related, so I exported the whole box.